pipeline.yaml

```diff
@@ -30,60 +35,73 @@ data:
-    # Note that arbitrary transformations should go into the transform step
+    #
+    # FIXME::OPTIONAL: Adjust the train/validation/test split ratios below.
+    #
+    split_ratios: [0.75, 0.125, 0.125]
+    #
+    #  FIXME::OPTIONAL: Specifies the method to use to "post-process" the split datasets. Note that
+    #                   arbitrary transformations should go into the transform step.
     post_split_method: steps.split.process_splits
   transform:
-    # Specifies the method that defines the data transformations to apply during model inference
+    #
+    #  FIXME::OPTIONAL: Specifies the method that defines an sklearn-compatible transformer, which
+    #                   applies input feature transformation during model training and inference.
     transformer_method: steps.transform.transformer_fn
   train:
     using: estimator_spec
     # Specifies the method that defines the estimator type and parameters to use for model training
     estimator_method: steps.train.estimator_fn
   evaluate:
-    # Sets performance thresholds that a trained model must meet in order to be eligible for
-    # registration to the MLflow Model Registry
-    validation_criteria:
-      - metric: root_mean_squared_error
-        threshold: 10
-      - metric: mean_absolute_error
-        threshold: 50
-      - metric: weighted_mean_squared_error
-        threshold: 20
+    #
+    # FIXME::OPTIONAL: Sets performance thresholds that a trained model must meet in order to be
+    #                  eligible for registration to the MLflow Model Registry.
+    #
+    # validation_criteria:
+    #   - metric: root_mean_squared_error
+    #     threshold: 10
   register:
-    # Specifies the name of the Registered Model to use when registering a trained model to
-    # the MLflow Model Registry
-    model_name: "taxi_fare_regressor"
+    #
+    # FIXME::REQUIRED: Specifies the name of the Registered Model to use when registering a trained
+    #                  model to the MLflow Model Registry.
+    #
+    model_name: ""
     # Indicates whether or not a model that fails to meet performance thresholds should still
     # be registered to the MLflow Model Registry
     allow_non_validated_model: false
   predict:
-    # Specifically define the model URI to use in batch scoring here or use the latest model
-    # registered from the training DAG
+    #
+    # FIXME::OPTIONAL: specify the model URI in batch scoring. If empty, the latest model registered
+    #                  from the training step will be used.
     # model_uri: "models/model.pkl"
-    # Specify the output path of the scored data from predict
-    output_location: {{SCORED_OUTPUT_DATA_LOCATION}}
-    # Specify the output format of the scored data from predict
-    output_format: {{SCORED_OUTPUT_DATA_FORMAT|default('parquet')}}
+    #
+    # FIXME::OPTIONAL: Specify the output path of the scored data from predict.
+    #
+    # output_location: {{SCORED_OUTPUT_DATA_LOCATION}}
```

Comment:
Output location is required for batch scoring, right? In that case, I think we should uncomment this.